src/T2DMap.cpp

```diff
@@ -3142,6 +2881,267 @@ void T2DMap::mousePressEvent(QMouseEvent* event)
         mPopupMenu = true;
         popup->popup(mapToGlobal(event->pos()));
     }
+}
+
+bool T2DMap::event(QEvent* event)
+{
+    // NOTE: key events aren't being forwarded to T2DMap because the widget
+    // currently never has focus because it's more comfortable for the user
+    // to always have focus on the command line. If this were to be changed some
+    // day the setFocusPolicy() calls in the constructor need to be uncommented
+
+    if (event->type() == QEvent::KeyPress) {
+//        auto* ke = static_cast<QKeyEvent*>(event);
+//        if (ke->key() == Qt::Key_Delete ) {
+//            if (mCustomLineSelectedRoom != 0  ) {
+//                if (mpMap->rooms.contains(mCustomLineSelectedRoom)) {
+//                    TRoom * pR = mpMap->rooms[mCustomLineSelectedRoom];
+//                    if (pR->customLines.contains( mCustomLineSelectedExit)) {
+//                        pR->customLines.remove(mCustomLineSelectedExit);
+//                        repaint();
+//                        mCustomLineSelectedRoom = 0;
+//                        mCustomLineSelectedExit = "";
+//                        mCustomLineSelectedPoint = -1;
+//                        return QWidget::event(event);
+//                    }
+//                }
+//            }
+//        }
+    } else if (event->type() == QEvent::Resize) { // Tweak the room selection widget to fit
+        resizeMultiSelectionWidget();
+    }
+    return QWidget::event(event);
+}
+
+void T2DMap::mousePressEvent(QMouseEvent* event)
+{
+    if (!mpMap) {
+        return;
+    }
+    mudlet::self()->activateProfile(mpHost);
+    mNewMoveAction = true;
+    if (event->buttons() & Qt::LeftButton) {
```

Comment:
Not sure if it has a baring on things or not but `QMouseEvent::buttons()` includes ALL the buttons that are down on entry to the event - so the code here will be run if any other buttons are down besides the left one - e.g. the right one -  so, for instance if there is a drag-and-select going on with the left-button **held down** and the right one is tapped, it will raise a `mousePressEvent` and the code in this `if` **will** be run. We may need/want to change this:
```suggestion
    if (event->button() & Qt::LeftButton) {
```

Reply:
While I know git's diff output says I added this, really all I did was copy and paste the right-click menu code from the press event to the release event. I think there was just enough code in common to confuse diff. That being said I think this is the only one that uses buttons instead of button, so another PR to change that may be a good idea.